modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaAssetEntryAssetCategoriesCheck.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.check.util.SourceUtil;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nícolas Moura
+ */
+public class UpgradeJavaAssetEntryAssetCategoriesCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		String newContent = _replaceAddOrDeleteAssetCategories(content);
+
+		newContent = _replaceAddOrDeleteAssetCategory(newContent);
+
+		if (!content.equals(newContent)) {
+			newContent = JavaSourceUtil.addImports(
+				newContent,
+				"com.liferay.asset.entry.rel.service." +
+					"AssetEntryAssetCategoryRelLocalService",
+				"org.osgi.service.component.annotations.Reference");
+
+			newContent = StringUtil.replaceLast(
+				newContent, CharPool.CLOSE_CURLY_BRACE,
+				"\n\t@Reference\n\tprivate " +
+					"AssetEntryAssetCategoryRelLocalService\n\t\t" +
+						"_assetEntryAssetCategoryRelLocalService;\n\n}");
+		}
+
+		return newContent;
+	}
+
+	private String _replaceAddOrDeleteAssetCategories(String content) {
+		String newContent = content;
+
+		Matcher addOrDeleteAssetEntryAssetCategoriesMatcher =
```

Comment:
You can just call the variable `matcher`. You only need to give it a different name when you are dealing with multiple `Matcher` objects. Plus, it makes the expressions, in my opinion, weirdly long.

Reply:
Good point, thx